### PR TITLE
Web UI code refactor on health page

### PR DESCRIPTION
**Pull request by imcotton**
### What does this PR do?

- Giving type structure of API calling result to endpoint `/health`, gains static typing.
- Reduce `subscription` in favor of `async` pipe.
- Having field property initialized earlier.
- Having `private readonly` modifier when it's possible.
- Setting change detection strategy as `OnPush`.
- Window resize event hooks on debouncing, and call `unsubscribe` to avoid memory leaking.
-  Passing unit tests.

### Motivation

Best practice of Angular, TypeScript and RxJS.

### Additional Notes

Any concerns, feedbacks are welcome.

### CODE REVIEW

File: webui/src/app/charts/bar-chart/bar-chart.component.ts [hunk 1]
```diff
@@ -2,21 +2,39 @@ import {
...
+}
+
 @Component({
   selector: 'app-bar-chart',
   templateUrl: './bar-chart.component.html'
 })
-export class BarChartComponent implements OnInit, OnChanges {
-  @Input() value: any;
+export class BarChartComponent implements OnInit, OnDestroy {
+  @Input() set value(data: DataModel[] | null) {
```
Comment (Slashgear) [inline webui/src/app/charts/bar-chart/bar-chart.component.ts]:
```suggestion
  @Input() set value(data?: DataModel[]) {
```

File: webui/src/app/charts/line-chart/line-chart.component.ts [hunk 2]
```diff
@@ -52,9 +54,22 @@ export class LineChartComponent implements OnChanges, OnInit {
   margin = { top: 40, right: 40, bottom: 60, left: 60 };
   loading = true;
 
+  private readonly resize$$ = this.windowService.resizeDebounce$.subscribe(
+    w => {
```
Comment (Slashgear) [inline webui/src/app/charts/line-chart/line-chart.component.ts]:
```suggestion
    () => {
```

File: webui/src/app/charts/bar-chart/bar-chart.component.ts (hunk 1 above)
Comment (imcotton) [reply webui/src/app/charts/bar-chart/bar-chart.component.ts]: This is case before the changing, however in setter case with `async` input, `null` is exact the first value been passed in.

File: webui/src/app/charts/line-chart/line-chart.component.ts (hunk 2 above)
Comment (imcotton) [reply webui/src/app/charts/line-chart/line-chart.component.ts]: Good catch, will remove this unused parameter.

**Comment by ldez** [comment]
Thanks for your contribution :+1: 

As you can see the WebUI has been drop temporary, we are working on a new WebUI.

I will close your PR but I hope you will help us to improve the future WebUI.

Thank you again.

**Pull request closed without merging**